fastlane/spec/unused_options_spec.rb

```diff
@@ -4,10 +4,12 @@
       let (:all_exceptions) { %w(pilot appstore cert deliver gym match pem produce scan sigh snapshot supply testflight mailgun testfairy ipa import_from_git hockey deploygate crashlytics artifactory appledoc slather screengrab download_dsyms notification frameit set_changelog register_device register_devices latest_testflight_build_number app_store_build_number sh swiftlint) } # rubocop:disable Metrics/LineLength
 
       Fastlane::ActionsList.all_actions do |action, name|
-        next unless action.available_options.kind_of?(Array)
-        next unless action.available_options.last.kind_of?(FastlaneCore::ConfigItem)
-
         it "No unused parameters in '#{name}'" do
+          allow(FastlaneCore::FastlaneFolder).to receive(:path).and_return("./fastlane")
+
+          next unless action.available_options.kind_of?(Array)
+          next unless action.available_options.last.kind_of?(FastlaneCore::ConfigItem)
```

Comment:
OK, I see what's happening. The looping done by `Fastlane::ActionsList.all_actions do |action, name|` happens during spec definition time, well before the `before` blocks get evaluated during execution. That's why the calls to `next` need to be moved inside of the body of the spec example... otherwise they are evaluated before the mocking can take place.

This seems like a reasonable compromise 👍